src/app/analyze/analyze-overview/analyze-overview.component.ts

```diff
@@ -11,26 +11,36 @@ import { ContextService } from '../../shared/context.service';
   templateUrl: 'analyze-overview.component.html',
   styleUrls: ['./analyze-overview.component.scss']
 })
-export class AnalyzeOverviewComponent {
+export class AnalyzeOverviewComponent implements OnInit {
 
   imgLoaded: Boolean = false;
 
   space: Space;
 
   constructor(
+    private route: ActivatedRoute,
     private router: Router,
     context: ContextService,
+    private spaceService: SpaceService,
     private broadcaster: Broadcaster
   ) {
-    context.current.subscribe(val => this.space = val.space);
+  }
+
+  ngOnInit() {
+    this.route.params
+      .switchMap((params: Params) => this.spaceService.getSpaceByName(params['entity'], params['space']))
+      .subscribe((space: Space) => this.space = space);
```

Comment:
I've made these changes, tested them locally, and pushed them. It works for the most part. There are some template errors reported in the console occasionally, which I might chalk down to some race conditions. They don't seem to crash the UI, but it't too early to tell.

![image](https://cloud.githubusercontent.com/assets/792234/23686043/5449a8a0-03cd-11e7-8512-bf803872ebec.png)